oat-sa/extension-tao-testqti
# Feature/tao 7617/persist highlighted text

The main PR for https://oat-sa.atlassian.net/browse/TAO-7617
Please read Jira ticket comments first to understand what the code does.

Depends on the following 2 PRs being merged:
- [x] https://github.com/oat-sa/tao-core/pull/2041
- [x] https://github.com/oat-sa/extension-tao-itemqti/pull/1251

To test this PR:
In a standard TAO install, create or import a test with some text items and enable the highlighter plugin. The highlighter should behave normally (keeps your item text highlighted when you move between items in a test, but loses highlights when you stop and resume in another browser).

Next, create some text stimuli (a QTI package will be provided in the Jira ticket) and import them (you will need `extension-tao-mediamanager` installed for this). Create more text items and include 1 or more stimuli, even the same stimulus across several items. Create a new test with these items and enable the highlighter tool again.

When you take the test with the stimuli, the highlights you make on a shared stimulus should appear every time that stimulus appears. The item text highlights should behave as they did before. Also, if you leave the test and resume it on another machine or browser, all the highlights should persist.

Additional Notes:
There is a technical limitation: you will observe that the highlighter tool is not able to span item text and stimulus text in one movement.
